### PR TITLE
feat: Audit GetDecisions

### LINKED ISSUE #885: Audit Events: GetDecision
Parent: #719 

Whimsical: https://whimsical.com/audit-HkQXHSMxu8xaWZUznhFCPW

Add audit logs for `GetDecision`

This is by far the most complicated audit event we'll have so far. The schema should be as described in the whimsical document and as defined below (the whimsical is a living document so if there are any differences it should take priority)

```jsonc
{
  "object": {
    "type": "entity_object",
    "id": "uuid", // entity chain id + resource id in the request
    "name": "string - optional",
    "attributes": {
      "attrs": [], // string of fqns in resource attributes
      "permissions": [] // - only for user_objects
    }
  },
  "action": {
    "type": "string - [create, read, update, delete, triggered, encrypt, decrypt, access_denied, access_modified, data_shared, error, block, ignore, override, cancel, unwrap, rewrap, digest, key_decrypt, key_sign, private_key_wrap]",
    "result": "string - [success, error, (more to be added here)]"
  },
  "owner": {
    "id": "ad484b11-c3ee-44d1-b552-7ee9bdcf33b0",
    "orgId": "ad484b11-c3ee-44d1-b552-7ee9bdcf33b0",
  },
  "actor": {
    "id": "string", // entity chain id
    "attributes": {
      "permissions": [],
      "attrs": [] // https://github.com/opentdf/platform/blob/main/service/authorization/authorization.proto#L200-L217
    }
  },
  "eventMetaData": {
    "chains": {
      "entityChainId": "ec1",
      "resourceAttributesId": "attr-set-2",
      "decision": "DECISION_DENIED",
      "entities": [
        {
          "id": "bob@example.com",
          "decision": "DECISION_PERMIT",
          "entitlements": [
            // fqns..... 
          ]
        },
        {
          "id": "bob2@example.com",
          "decision": "DECISION_DENIED",
          "error": "what failed??"
        }
      ]
    }
  },
  "clientInfo": {
    "userAgent": "string",
    "platform": "string - secure reader, bp, etc..",
    "requestIp": "string",
  },
  "diff": "show's what changed in the object - only needed for update actions",
  "timestamp": "date"
}
```

### References

Useful references:
- [Decision Request](https://github.com/opentdf/platform/blob/529fb0ec775eca93f8cdd83388eba950a5e81bba/service/authorization/authorization.proto#L47-86)
- [Decision Response](https://github.com/opentdf/platform/blob/529fb0ec775eca93f8cdd83388eba950a5e81bba/service/authorization/authorization.proto#L107-L129)
- [Get Entitlements Request](https://github.com/opentdf/platform/blob/529fb0ec775eca93f8cdd83388eba950a5e81bba/service/authorization/authorization.proto#L159-L176)
- [Get Entitlements Response](https://github.com/opentdf/platform/blob/529fb0ec775eca93f8cdd83388eba950a5e81bba/service/authorization/authorization.proto#L199-L218)

### PR DESCRIPTION
**Pull request by pbacon-blaber**
- adds audit event for authorization.GetDecisions
- closes #885

### CODE REVIEW

File: sdk/audit/metadata_adding_interceptor.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/realip"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+const (
+	requestIDHeaderKey = "x-request-id"
+	requestIPHeaderKey = "x-forwarded-request-ip"
+	actorIDHeaderKey   = "x-forwarded-actor-id"
+)
```
Comment (pflynn-virtru) [inline sdk/audit/metadata_adding_interceptor.go:16]: use the constants from context_keys

File: service/internal/logger/audit/constants.go [hunk 2]
```diff
@@ -1,5 +1,18 @@
...
+)
+
+type RequestHeader string
+
+const (
+	UserAgentHeaderKey RequestHeader = "user-agent"
+	RequestIDHeaderKey RequestHeader = "x-request-id"
+	RequestIPHeaderKey RequestHeader = "x-forwarded-request-ip"
+	ActorIDHeaderKey   RequestHeader = "x-forwarded-actor-id"
+)
```
Comment (pflynn-virtru) [inline service/internal/logger/audit/constants.go:14]: if there is no circular dependencies, then use sdk constants
`sdkAudit "github.com/opentdf/platform/sdk/audit"`

**Review by pflynn-virtru — APPROVED** [review]

File: sdk/audit/metadata_adding_interceptor.go (hunk 1 above)
Comment (pbacon-blaber) [reply sdk/audit/metadata_adding_interceptor.go:16]: good catch, replaced

File: service/internal/logger/audit/constants.go (hunk 2 above)
Comment (pbacon-blaber) [reply service/internal/logger/audit/constants.go:14]: these actually weren't being referenced anywhere as i had migrated everything off of them so I just deleted them

**Review by pflynn-virtru — APPROVED** [review]

**Review by strantalis — APPROVED** [review]

**Pull request merged**